### PR TITLE
[mono][tests] Fix FullAOT DivToMul test

### PR DESCRIPTION
**Pull request by LeVladIonescu**
LLVM is optimizing divisions by -1 so that a neg operation is executed instead of a div operation.
When using NaN values, neg operation is returning a different bit configuration for NaN compared to the one returned by div. 
This change is verifying that the arguments are valid NaNs, it doesn't affect the test's purpose.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=78816) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=78816) before we can accept your contribution.<br/><br/>:x: LeVladIonescu [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=78816)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=78816) it.</sub>

**Review by EgorBo — APPROVED** [review]

**Review by kotlarmilos — APPROVED** [review]
lgtm :)

File: src/tests/JIT/opt/InstructionCombining/DivToMul.cs [hunk 1]
```diff
@@ -160,6 +160,13 @@ private static void TestNotPowOfTwo_Double(double x)
     [MethodImpl(MethodImplOptions.NoInlining)]
     private static void AssertEquals(float expected, float actual)
     {
+        if (Single.IsNaN(expected) && Single.IsNaN(actual))
+        {
+            // There can be multiple configurations for NaN values
+            // verifying that this values are NaNs should be enough
```
Comment (vargaz) [inline src/tests/JIT/opt/InstructionCombining/DivToMul.cs:166]: Typo.

File: src/tests/JIT/opt/InstructionCombining/DivToMul.cs [hunk 2]
```diff
@@ -172,6 +179,13 @@ private static void AssertEquals(float expected, float actual)
     [MethodImpl(MethodImplOptions.NoInlining)]
     private static void AssertEquals(double expected, double actual)
     {
+        if (Double.IsNaN(expected) && Double.IsNaN(actual))
+        {
+            // There can be multiple configurations for NaN values
+            // verifying that this values are NaNs should be enough
+            return;
```
Comment (vargaz) [inline src/tests/JIT/opt/InstructionCombining/DivToMul.cs:186]: Ditto.

**Pull request merged**